src/doppler/sinks/syslogwriter/https_writer_test.go

```diff
@@ -83,7 +91,76 @@ var _ = Describe("HttpsWriter", func() {
 			parsedTime, err := time.Parse(time.RFC3339, "2006-01-02T15:04:05Z")
 			_, err = w.Write(standardErrorPriority, []byte("Message"), "just a test", "TEST", parsedTime.UnixNano())
 			Expect(err).ToNot(HaveOccurred())
+		})
 
+		Describe("idle connection behavior", func() {
+			var (
+				receivedChan chan []byte
+				waitChan     chan struct{}
+				listener     *historyListener
+			)
+
+			BeforeEach(func() {
+				receivedChan = make(chan []byte, 1)
+				waitChan = make(chan struct{})
+
+				l, err := net.Listen("tcp", "127.0.0.1:0")
+				Expect(err).NotTo(HaveOccurred())
+
+				listener = &historyListener{Listener: l}
+				server.Listener = listener
+
+				slowHandler := func(_ http.ResponseWriter, r *http.Request) {
+					body, _ := ioutil.ReadAll(r.Body)
+					receivedChan <- body
+
+					<-waitChan
+					r.Body.Close()
+				}
+				serveMux.HandleFunc("/wait/", slowHandler)
+			})
+
+			It("only keeps one idle connection", func() {
+				outputUrl, err := url.Parse(server.URL + "/wait/")
+				Expect(err).NotTo(HaveOccurred())
+
+				w, err := syslogwriter.NewHttpsWriter(outputUrl, "appId", true)
+				Expect(err).NotTo(HaveOccurred())
+
+				completedReqs := make(chan struct{}, 2)
+				blockingRequest := func() {
+					defer GinkgoRecover()
+					_, err := w.Write(standardErrorPriority, []byte("Message"), "just a test", "TEST", time.Now().UnixNano())
+					Expect(err).NotTo(HaveOccurred())
+					Eventually(completedReqs).Should(BeSent(struct{}{}))
+				}
+
+				for i := 0; i < 2; i++ {
```

Comment:
It looks like this code is duplicated below from [Line 151](https://github.com/cloudfoundry/loggregator/pull/70/files#diff-1faabae3338304697e132f3655dab4feR151) to [Line 159] (https://github.com/cloudfoundry/loggregator/pull/70/files#diff-1faabae3338304697e132f3655dab4feR159)